### PR TITLE
refactor(threed): push args in docstring to child class

### PR DESCRIPTION
**Pull request by swheaton**
## What changes are proposed in this pull request?

In service of `Explicit is better than implicit`, I like putting all inherited args in the child class docstring and init up front. This way you see exactly how to create an instance and don't have to hop back and forth in the family tree.

The flip side is that there is a lot of duplication and if an arg is added to `Object3D` then it has to be pushed to all children. I would argue this is an acceptable few-time cost in exchange for more readable code and user documentation.

`**kwargs` is interesting if the args can truly be dynamic, in my opinion.

Open for discussion.

## How is this patch tested? If it is not, please explain why.

unit tests still fine

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
  - Improved lighting customization with added parameters for visibility, position, scale, and orientation in various light classes.
  - Introduced opacity control for different material types, enabling users to adjust transparency.
  - Updated 3D objects and shapes to include customizable visibility, position, scale, and orientation settings.

- **Refactor**
  - Enhanced constructor parameter lists for lights, materials, meshes, point clouds, and 3D shapes to provide advanced customization options.

### CODE REVIEW

File: fiftyone/core/threed/pointcloud.py
Comment (brimoor) [inline fiftyone/core/threed/pointcloud.py:32]: `name` is defined twice in this docstring. I think this one needs to be deleted (also indentation is borked)

**Review by brimoor — APPROVED** [review]
LGTM after fixing the one `name` issue I flagged.

And +1 for explicit parameter documentation! Tedious but very useful for end users 🤗 

@sashankaryal should probably approve too before merging as code owner

**Review by sashankaryal — APPROVED** [review]
lgtm!! ⚡

**Pull request merged**
